src/Rings/mpoly-ideals.jl

```diff
@@ -531,6 +531,23 @@ function primary_decomposition(I::T; algorithm::Symbol=:GTZ, cache::Bool=true) w
   end::Vector{Tuple{T,T}}
 end
 
+function primary_decomposition(
+    I::MPolyIdeal{T}; 
+    algorithm::Symbol=:GTZ, cache::Bool=true
+  ) where {U<:Union{nf_elem, <:Hecke.NfRelElem}, T<:MPolyRingElem{U}}
```

Comment:
The signature needs to be narrowed down:
```
julia> K, a = number_field([x - 1, x - 2]);

julia> Kt, t = K["t"];

julia> L, b = number_field(t - 1, "b");

julia> M, = number_field(t - 1, "b");

julia> Mx, = polynomial_ring(M, 2);

julia> primary_decomposition(ideal(Mx, [gen(Mx, 1)]))
ERROR: MethodError: no method matching _flatten_ring(::MPolyQuoRing{AbstractAlgebra.Generic.MPoly{NfAbsNSElem}})

Closest candidates are:
```

Reply:
It seems that there is another type `NfAbsNSElem` lurking around which I was not aware of. We'll settle this next week, I guess.

Reply:
There is also a fourth type. It was not clear to me whether this PR was intended to handle all types or just towers of simple extensions?

Reply:
I don't know either. I think we're fine with the towers of simple extensions for the moment. The rest can be done in a later step. I will leave a TODO comment in the code.

Reply:
I implemented all types of number fields now. For the moment, I still use the `.data` field of the elements. If there is a better and systematic way to get that information, then we can (and probably should) change that. Otherwise, I think, it can stay like this for now. If the internals of the number fields change in a way that this is affected, we will see it in the tests failing.